src/extras/PMREMGenerator.js

```diff
@@ -0,0 +1,832 @@
+	cubeUVRenderTarget.texture.name = 'PMREM.cubeUv';
+	cubeUVRenderTarget.scissorTest = true;
+	return cubeUVRenderTarget;
+
+}
+
+function _setViewport( x, y, width, height ) {
+
+	var invDpr = 1.0 / _renderer.getPixelRatio();
+	x *= invDpr;
+	y *= invDpr;
+	width *= invDpr;
+	height *= invDpr;
+	_renderer.setViewport( x, y, width, height );
+	_renderer.setScissor( x, y, width, height );
+
+}
+
+function _applyPMREM( cubeUVRenderTarget ) {
+
+	var autoClear = _renderer.autoClear;
+	_renderer.autoClear = false;
+
+		for ( var i = 1; i < TOTAL_LODS; i ++ ) {
+
+		var sigma = Math.sqrt(
+			_sigmas[ i ] * _sigmas[ i ] -
+		_sigmas[ i - 1 ] * _sigmas[ i - 1 ] );
+		var poleAxis =
+		_axisDirections[ ( i - 1 ) % _axisDirections.length ];
+		_blur( cubeUVRenderTarget, i - 1, i, sigma, poleAxis );
+
+	}
+
+	_renderer.autoClear = autoClear;
+
+}
+
+/**
+ * This is a two-pass Gaussian blur for a cubemap. Normally this is done
+ * vertically and horizontally, but this breaks down on a cube. Here we apply
+ * the blur latitudinally (around the poles), and then longitudinally (towards
+ * the poles) to approximate the orthogonally-separable blur. It is least
+ * accurate at the poles, but still does a decent job.
+ */
+function _blur( cubeUVRenderTarget, lodIn, lodOut, sigma, poleAxis ) {
+
+	_halfBlur(
+		cubeUVRenderTarget,
+		_pingPongRenderTarget,
+		lodIn,
+		lodOut,
+		sigma,
+		'latitudinal',
+		poleAxis );
+
+	_halfBlur(
+		_pingPongRenderTarget,
+		cubeUVRenderTarget,
+		lodOut,
+		lodOut,
+		sigma,
+		'longitudinal',
+		poleAxis );
+
+}
+
+function _halfBlur( targetIn, targetOut, lodIn, lodOut, sigmaRadians, direction, poleAxis ) {
+
+	if ( direction !== 'latitudinal' && direction !== 'longitudinal' ) {
+
+		console.error(
+			'blur direction must be either latitudinal or longitudinal!' );
+
+	}
+
+	// Number of standard deviations at which to cut off the discrete approximation.
+	var STANDARD_DEVIATIONS = 3;
+
+	var blurScene = new Scene();
+	blurScene.add( new Mesh( _lodPlanes[ lodOut ], _blurMaterial ) );
+	var blurUniforms = _blurMaterial.uniforms;
+
+	var pixels = _sizeLods[ lodIn ] - 1;
+	var radiansPerPixel = isFinite( sigmaRadians ) ? Math.PI / ( 2 * pixels ) : 2 * Math.PI / ( 2 * MAX_SAMPLES - 1 );
+	var sigmaPixels = sigmaRadians / radiansPerPixel;
+	var samples = isFinite( sigmaRadians ) ? 1 + Math.floor( STANDARD_DEVIATIONS * sigmaPixels ) : MAX_SAMPLES;
+
+	if ( samples > MAX_SAMPLES ) {
+
+		console.warn( `sigmaRadians, ${
+			sigmaRadians}, is too large and will clip, as it requested ${
+			samples} samples when the maximum is set to ${MAX_SAMPLES}` );
+
+	}
+
+	var weights = [];
+	var sum = 0;
+	for ( var i = 0; i < MAX_SAMPLES; ++ i ) {
+
+		var x = i / sigmaPixels;
+		var weight = Math.exp( - x * x / 2 );
+		weights.push( weight );
+		if ( i == 0 ) {
+
+				 sum += weight;
+
+		} else if ( i < samples ) {
+
+				sum += 2 * weight;
+
+		}
+
+	}
+	weights = weights.map( function ( w ) { return w / sum } );
```

Comment:
@elalish also changed this line:

```js
weights = weights.map( w => w / sum );
```

Reply:
One of these days I'll learn the rest of JS's shortcuts...

Reply:
I actually undid the shortcut 😅

Twice:
https://github.com/mrdoob/three.js/blob/dev/src/extras/PMREMGenerator.js#L545-L549